sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/VolumeEvaluationStage.java

```diff
@@ -0,0 +1,70 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.offer.MesosResource;
+import com.mesosphere.sdk.offer.ResourceRequirement;
+import org.apache.mesos.Protos;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.Resource.DiskInfo;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * This class evaluates an offer against a given {@link com.mesosphere.sdk.offer.OfferRequirement}, ensuring that it
+ * contains an appropriately-sized volume, and creating any necessary instances of
+ * {@link com.mesosphere.sdk.offer.ReserveOfferRecommendation},
+ * {@link com.mesosphere.sdk.offer.UnreserveOfferRecommendation}, and
```

Comment:
Once a volume is created we should never change its size.  So UNRESERVE should never occur for disks outside of the `ResourceCleaner` or similar.